Adjust snaps insights documentation (#615, #617) (#616)
MetaMask/metamask-docs: docs/guide/snaps-exports.md

### Returns

```typescript
import { Component } from '@metamask/snaps-ui';

type OnTransactionHandlerReturn = Promise<OnTransactionResponse>;

interface OnTransactionResponse {
  content: Component;
}
```

> - `onTransactionResponse` - The `content` object returned by the snap will be displayed using Custom UI alongside the confirmation for the transaction that `onTransaction` was called with. Keys and values will be displayed in the order received, with each key rendered as a title and each value rendered as a string.

Reviewer: Custom UI is mentioned only at https://github.com/MetaMask/metamask-docs/blob/main/docs/guide/snaps.md#custom-ui-in-metamask-using-a-defined-set-of-components from what I can tell - mentioning it without defining it (and referring to the definition) may confuse the reader. Adding that section would be awesome, but otherwise maybe we can omit mentioning it for now?

Reviewer: Ah, just noticed #618 :muscle:

Author: 100% agree. But that link wouldn't be valid until the custom ui PR is merged.

Author: Resolved by doing one PR for both tasks

### Returns

```typescript
import { Component } from '@metamask/snaps-ui';

type OnTransactionHandlerReturn = Promise<OnTransactionResponse>;

interface OnTransactionResponse {
  content: Component;
}
```

> - `onTransactionResponse` - The `content` object returned by the snap will be displayed using [Custom UI](./snaps-concepts.html#custom-ui) alongside the confirmation for the transaction that `onTransaction` was called with. Keys and values will be displayed in the order received, with each key rendered as a title and each value rendered as a string.

Reviewer 2: ```suggestion
- `onTransactionResponse` - The `content` object returned by the snap will be displayed using [Custom UI](./snaps-concepts.html#custom-ui) alongside the confirmation for the transaction that `onTransaction` was called with.
```
This doesn't apply anymore

### Examples

#### Typescript

```typescript

> import { OnTransactionHandler } from "@metamask/snap-types";
> import { panel, heading, text } from "@metamask/snaps-ui";

Reviewer 3: ```suggestion
import { OnTransactionHandler } from '@metamask/snap-types';
import { panel, heading, text } from '@metamask/snaps-ui';
```

export const onTransaction: OnTransactionHandler = async ({
  transaction,
  chainId,
}) => {
  const insights = /* Get insights */;
  return {
    content: panel([

>       heading("My Transaction Insights"),
>       text("Here are the insights:"),

Reviewer 3: ```suggestion
      heading('My Transaction Insights'),
      text('Here are the insights:'),
```

      ...(insights.map((insight) => text(insight.value)))
    ])
  };
};
```

#### Javascript

```js

> import { panel, heading, text } from "@metamask/snaps-ui";

Reviewer 3: ```suggestion
import { panel, heading, text } from '@metamask/snaps-ui';
```

module.exports.onTransaction = async ({
  transaction,
  chainId,
}) => {
  const insights = /* Get insights */;
  return {
    content: panel([

>       heading("My Transaction Insights"),
>       text("Here are the insights:"),

Reviewer 3: ```suggestion
      heading('My Transaction Insights'),
      text('Here are the insights:'),
```

Author: Why are these not caught by the linter? Any way to make it happen?
